test/app-luatest/gh_9806_http_client_version_test.lua

```diff
@@ -0,0 +1,100 @@
+    treegen.write_script(dir, script_name, http_request_script)
+
+    local args = {script_name}
+    local opts = {nojson = true, stderr = true}
+    local env = {HTTP_SERVER_PORT = g.server:name().port}
+    env.PROTO = 'http'
+    env.HTTP_VERSION = nil
+    local res = justrun.tarantool(dir, env, args, opts)
+    t.assert_str_contains(res.stderr, 'GET / HTTP/1.1')
+    env.HTTP_VERSION = '1.0'
+    local res = justrun.tarantool(dir, env, args, opts)
+    t.assert_str_contains(res.stderr, 'GET / HTTP/1.0')
+    env.HTTP_VERSION = '1.1'
+    local res = justrun.tarantool(dir, env, args, opts)
+    t.assert_str_contains(res.stderr, 'GET / HTTP/1.1')
+    env.HTTP_VERSION = '2'
+    local res = justrun.tarantool(dir, env, args, opts)
+    t.assert_str_contains(res.stderr, 'Connection: Upgrade, HTTP2-Settings')
+    env.HTTP_VERSION = '2-tls'
+    local res = justrun.tarantool(dir, env, args, opts)
+    t.assert_str_contains(res.stderr, 'GET / HTTP/1.1')
+    env.HTTP_VERSION = '2-prior-knowledge'
+    local res = justrun.tarantool(dir, env, args, opts)
+    t.assert_str_contains(res.stderr, '[HTTP/2] [1] OPENED stream')
+end
+
+g.test_https_version = function()
+    local dir = treegen.prepare_directory(g, {}, {})
+    local script_name = 'http_request.lua'
+    treegen.write_script(dir, script_name, http_request_script)
+
+    local args = {script_name}
+    local opts = {nojson = true, stderr = true}
+    local env = {HTTP_SERVER_PORT = g.server:name().port}
+    env.PROTO = 'https'
+    env.HTTP_VERSION = nil
+    local res = justrun.tarantool(dir, env, args, opts)
+    t.assert_str_contains(res.stderr, 'ALPN: curl offers h2,http/1.1')
+    env.HTTP_VERSION = '1.0'
+    local res = justrun.tarantool(dir, env, args, opts)
+    t.assert_str_contains(res.stderr, 'ALPN: curl offers http/1.1')
+    env.HTTP_VERSION = '1.1'
+    local res = justrun.tarantool(dir, env, args, opts)
+    t.assert_str_contains(res.stderr, 'ALPN: curl offers http/1.1')
+    env.HTTP_VERSION = '2'
+    local res = justrun.tarantool(dir, env, args, opts)
+    t.assert_str_contains(res.stderr, 'ALPN: curl offers h2,http/1.1')
+    env.HTTP_VERSION = '2-tls'
+    local res = justrun.tarantool(dir, env, args, opts)
+    t.assert_str_contains(res.stderr, 'ALPN: curl offers h2,http/1.1')
+    env.HTTP_VERSION = '2-prior-knowledge'
+    local res = justrun.tarantool(dir, env, args, opts)
+    t.assert_str_contains(res.stderr, 'ALPN: curl offers h2,http/1.1')
+end
+
+g.test_request_wrong_http_version = function()
+    local dir = treegen.prepare_directory(g, {}, {})
+    local script_name = 'http_request.lua'
+    treegen.write_script(dir, script_name, http_request_script)
+
+    local args = {script_name}
+    local opts = {nojson = true, stderr = true}
+    local env = {HTTP_SERVER_PORT = g.server:name().port}
+    env.PROTO = 'http'
+    env.HTTP_VERSION = '3.0'
+    local res = justrun.tarantool(dir, env, args, opts)
+    t.assert_equals(res.exit_code, 0)
+end
```

Comment:
We don't need to catch stderr output from the `verbose` option here, so it is not necessary to start the HTTP client in a separate process.

I would do something like the following.

```lua
local http_client = require('http.client').new()
local exp_err = <...>
t.assert_error_msg_equals(exp_err, function()
    http_client:get(<...>, {http_version = <...>})
end)
```

Reply:
Fixed